tests/common/plugins/fib.py

```diff
@@ -197,18 +205,26 @@ def fib_t1_lag(ptfhost, testbed):
         port6 = 6000 + vm_offset
 
         if 'spine' in v['properties']:
-            announce_routes(ptfip, port, "v4", podset_number, tor_number, tor_subnet_number,
-                            None, leaf_asn_start, tor_asn_start,
-                            local_ip, local_ipv6, router_type="spine")
-
-            announce_routes(ptfip, port6, "v6", podset_number, tor_number, tor_subnet_number,
-                            None, leaf_asn_start, tor_asn_start,
-                            local_ip, local_ipv6, router_type="spine")
+            routes_v4 = generate_routes("v4", podset_number, tor_number, tor_subnet_number,
+                                        None, leaf_asn_start, tor_asn_start,
+                                        local_ip, local_ipv6, router_type="spine")
+            routes_v6 = generate_routes("v6", podset_number, tor_number, tor_subnet_number,
+                                        None, leaf_asn_start, tor_asn_start,
+                                        local_ip, local_ipv6, router_type="spine")
+            announce_routes(ptfip, port, routes_v4 + routes_v6)
 
         elif 'tor' in v['properties']:
-            announce_routes(ptfip, port, "v4", podset_number, tor_number, tor_subnet_number,
-                            None, leaf_asn_start, tor_asn_start,
-                            local_ip, local_ipv6, router_type="tor")
+            routes_v4 = generate_routes("v4", podset_number, tor_number, tor_subnet_number,
+                                        None, leaf_asn_start, tor_asn_start,
+                                        local_ip, local_ipv6, router_type="tor")
+            announce_routes(ptfip, port, routes_v4)
+
+        if 'vips' in v:
+            routes_vips = []
+            for prefix in v["vips"]["ipv4"]["prefixes"]:
+                routes_vips.append((prefix, local_ip, v["vips"]["ipv4"]["asn"]))
+            announce_routes(ptfip, port, routes_vips)
```

Comment:
I am not sure we want to advertise to every tor about the vip. I remember topology has information which tor needs to advertise the vip.

Reply:
This will not advertise to every tor. It will only advertise to tor VMs that have configuration "vips" in topology definition file. 

For example, in ansible/vars/topo_t1-lag.yml, only two VMs have the "vips" configuration: ARISTA01T0, ARISTA03T0.

```
  ARISTA01T0:
    properties:
    - common
    - tor
    tornum: 1
    bgp:
      asn: 64001
      peers:
        65100:
        - 10.0.0.32
        - FC00::41
    interfaces:
      Loopback0:
        ipv4: 100.1.0.17/32
        ipv6: 2064:100::11/128
      Ethernet1:
        ipv4: 10.0.0.33/31
        ipv6: fc00::42/126
    bp_interface:
      ipv4: 10.10.246.17/24
      ipv6: fc0a::22/64
    vips:
      ipv4:
        prefixes:
           - 200.0.1.0/26
        asn: 64700
```

```
  ARISTA03T0:
    properties:
    - common
    - tor
    tornum: 3
    bgp:
      asn: 64003
      peers:
        65100:
        - 10.0.0.36
        - FC00::49
    interfaces:
      Loopback0:
        ipv4: 100.1.0.19/32
        ipv6: 2064:100::13/128
      Ethernet1:
        ipv4: 10.0.0.37/31
        ipv6: fc00::4a/126
    bp_interface:
      ipv4: 10.10.246.19/24
      ipv6: fc0a::26/64
    vips:
      ipv4:
        prefixes:
          - 200.0.1.0/26
        asn: 64700
```

The vips routes will only be advertised to these two tor VMs.